apps/sla/models.py

```diff
@@ -156,22 +190,47 @@ class SLAPolicy:
         "tracker": Tracker,
     }
 
-    def __init__(self, policy_desc):
-        self.name = policy_desc["name"]
-        self.description = policy_desc["description"]
-        self._init_conditions(policy_desc["conditions"])
-        self.sla = SLA(policy_desc["sla"])
-
-    def _init_conditions(self, conditions_desc):
-        """
-        the conditions need to be split entity-wise
-        """
-        self.conditions = {}
-        for entity, conditions in conditions_desc.items():
-            self.conditions[entity] = [
+    name = models.CharField(max_length=100)
+    description = models.TextField()
+    sla = models.ForeignKey(SLA, on_delete=models.CASCADE, related_name="policies")
+    condition_descriptions = models.JSONField(default=dict)
+    order = models.IntegerField()
+
+    class Meta:
+        # Order of SLA is important, so by default retrieve them using the order field
+        ordering = ["order"]
+
+    @classmethod
+    def create_from_description(self, policy_desc, order=None):
+        """Creates an SLA policy from a YAML description."""
+        name = policy_desc["name"]
+        description = policy_desc["description"]
+        sla = SLA.create_from_description(policy_desc["sla"])
+        sla.save()
+
+        if order is None:
+            # Order is implied by the number of already existing SLA policies
+            order = SLAPolicy.objects.count()
```

Comment:
This sounds to me a bit random (maybe it is not). The order must correspond to the vertical ordering of the SLA policies in the file as it is used for prioritization.

Reply:
They are loaded sequentially from the YAML in the same order they are defined, so the order is maintained this way.